### PR TITLE
Add day ahead model config

### PR DESCRIPTION
**Pull request by Sukh-P**
# Pull Request

## Description

Changes to run a separate day ahead PVNet model, which will be deployed alongside the existing intraday PVNet ones. To run the new model a new env variable called DAY_AHEAD_MODEL will need to be added and set to "True".  

Some linting has also been applied, will comment on actual changes for clarity

## How Has This Been Tested?

Added a test to the test_app file

## Checklist:

- [X] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pvnet_app/app.py [hunk 1]
```diff
@@ -108,17 +105,37 @@
         "save_gsp_sum": False,
         "verbose": False,
     },
-    
+}
+
+day_ahead_model_dict = {
```
Comment (Sukh-P) [inline pvnet_app/app.py:110]: New dict for model config

File: pvnet_app/app.py [hunk 2]
```diff
@@ -164,16 +182,27 @@ def app(
 
     if num_workers == -1:
         num_workers = os.cpu_count() - 1
-    if num_workers>0:
+    if num_workers > 0:
         # Without this line the dataloader will hang if multiple workers are used
-        dask.config.set(scheduler='single-threaded')
+        dask.config.set(scheduler="single-threaded")
+
+    day_ahead_model_used = os.getenv("DAY_AHEAD_MODEL", "false").lower() == "true"
```
Comment (Sukh-P) [inline pvnet_app/app.py:189]: Logging changes

File: pvnet_app/app.py [hunk 3]
```diff
@@ -164,16 +182,27 @@ def app(
...
+        logger.info(f"Using day ahead PVNet model")
 
     logger.info(f"Using `pvnet` library version: {pvnet.__version__}")
     logger.info(f"Using `pvnet_app` library version: {pvnet_app.__version__}")
     logger.info(f"Using {num_workers} workers")
-    logger.info(f"Using adjduster: {models_dict['pvnet_v2']['use_adjuster']}")
-    logger.info(f"Saving GSP sum: {models_dict['pvnet_v2']['save_gsp_sum']}")
-    
+
+    if day_ahead_model_used:
```
Comment (Sukh-P) [inline pvnet_app/app.py:198]: Logging changes

File: pvnet_app/app.py [hunk 4]
```diff
@@ -213,22 +242,24 @@ def app(
...
+
     # Preprocess the NWP data
     preprocess_nwp_data()
-    
+
     # ---------------------------------------------------------------------------
     # 2. Set up models
-    
+
+    if day_ahead_model_used:
```
Comment (Sukh-P) [inline pvnet_app/app.py:259]: Changing which model is used

File: pvnet_app/forecast_compiler.py [hunk 5]
```diff
@@ -36,8 +36,8 @@ def __init__(
         self, 
         model_name: str, 
         model_version: str, 
-        summation_name: str, 
-        summation_version: str, 
+        summation_name: str | None,
```
Comment (Sukh-P) [inline pvnet_app/forecast_compiler.py:39]: Update type hints

File: tests/test_app.py [hunk 6]
```diff
@@ -73,4 +74,81 @@ def test_app(
     # 318 GSPs * 16 time steps in forecast
     assert len(db_session.query(ForecastValueSQL).all()) == expected_forecast_results * 16
     assert len(db_session.query(ForecastValueLatestSQL).all()) == expected_forecast_results * 16
-    assert len(db_session.query(ForecastValueSevenDaysSQL).all()) == expected_forecast_results * 16
\ No newline at end of file
+    assert len(db_session.query(ForecastValueSevenDaysSQL).all()) == expected_forecast_results * 16
+
+
+def test_app_day_ahead_model(
```
Comment (Sukh-P) [inline tests/test_app.py:80]: New test

**Comment by peterdudfield** [comment]
Looks great, have your run the app locally? That might be the final step before we merge?

**Comment by Sukh-P** [comment]
> Looks great, have your run the app locally? That might be the final step before we merge?

Thanks, trying to get it to run locally currently, running into some errors related to the NWP zarr coords (seemingly unrelated to the changes I have made) that the CI doesn't seem to encounter so just digging into why that is

**Review by peterdudfield — APPROVED** [review]
Sukh tested this locally and it worked!

**Pull request merged**